test/CMakeLists.txt

```diff
@@ -200,7 +212,12 @@ else()
         target_link_libraries(gtest_zlib external_zlib)
     endif()
 
-    target_link_libraries(gtest_zlib zlibstatic GTest::GTest)
+    if(ZLIB_COMPAT_TESTS)
+        target_link_libraries(gtest_zlib zlib GTest::Main)
```

Comment:
Why switch between zlib and zlibstatic? Both libraries exist in zlib's CMake.

Reply:
There shouldn't be any reason to change this block of code... Even if there was, there should be at least a comment to make it obvious why changing static to "possibly" dynamic linkage is required or warranted.

Reply:
It's needed in order to be able to replace zlib-ng with zlib at runtime. Of course, this may be defined as a separate option, but ZLIB_COMPAT_TESTS=ON/ZLIBNG_ENABLE_TESTS=OFF is not too useful without it.

Reply:
I think adding comment would a good idea then, and like you said, you may need to take into account the value of `BUILD_SHARED_LIBS`.

Reply:
Ok, I've implemented this in the commit I pushed.